### PR TITLE
Hour of Code overview page

### PR DESCRIPTION
**Pull request by Erin007**
A new Hour of Code overview page.  This PR also include hoc_courses_narrow because there is a narrower space for the hoc_tiles with the side nav on this page. Given the narrower width, the cards only fit 3 across, so I added an Unplugged Conditionals tile to make a 3 by 3 grid. 

hour of code overview

### CODE REVIEW

**Comment by Erin007** [comment]
cc: @joshlory

**Comment by Bjvanminnen** [comment]
Not sure if there are other instances of doing stuff like this, but it feels weird to me to have a dashboard view that we expose to pegasus by via dashboardapi.

**Comment by tanyaparker** [comment]
- Let's make the purple title "Hour of Code for all ages"✅
- Get rid of Available courses✅
- Since it's 3 column, can we create a 9th card that links to the Conditionals Unplugged✅

**Comment by Erin007** [comment]
@Bjvanminnen teacher announcements work like this to my understanding. Do you have a preferred alternative?

**Comment by tanyaparker** [comment]
Also, don't know if you want to do this in the same PR but there is a component on the teacher's Courses page that links to code.org/hourofcode/all that we need to update to link to this page.✅

**Comment by Bjvanminnen** [comment]
I'm not sure. Might be worth chatting about this in person with you/me/Brendan.

**Comment by breville** [comment]
The cool new way to do this is to put the HAML in /shared/haml and have both dashboard and pegasus load it server-side.  To minimize risk and move quickly (we're going live next week with some rather big changes we just learnt about this morning) I suggested using the ajax technique which is already used by teacher-dashboard.  That said, the cool new way might actually work pretty easily; I can help @Erin007 try this shortly.

File: pegasus/sites.v3/code.org/public/hourofcode/overview.haml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  .coursesheader
+    %h2{style: 'margin-bottom: 0px'}= I18n.t('dashboard_landing_available_title')
+    %h3= I18n.t('dashboard_hoc_courses')
+    = I18n.t('dashboard_hoc_courses_desc_and_cta')
+  %br/
+
+.hoc-tiles-container
+
+.full-resource-block
+  %img.side-image.left.col-50-student{src: "/images/fit-485/student/hour_of_code.jpg"}
```
Comment (tanyaparker) [inline pegasus/sites.v3/code.org/public/hourofcode/overview.haml]: The image of the girl is stretched vertically. Needs to be 388px width?

File: pegasus/sites.v3/code.org/public/hourofcode/overview.haml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      %h2.resource-title More Activities from Partners
+      The Hour of Code website has over 100 computer science activities from Code.org and partners for all ages and experience levels.
+    %a{href: "/learn"}
+      %button.orange-button Go
+
+:javascript
+  $(document).ready(function () {
+    $(".hoc-tiles-container").load( "/dashboardapi/hoc_courses_narrow", function() {
+    });
+  });
```
Comment (Bjvanminnen) [inline pegasus/sites.v3/code.org/public/hourofcode/overview.haml]: If we do stick with this approach, we should move this javascript into a JS file.

**Comment by breville** [comment]
Ah, we hit a snag with the cool new way, which is that a partial in question (https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/views/shared/_course_tall_block.haml) uses the dashboard loc pipeline to render strings.  So unfortunately we need to stick with the ajax approach so that dashboard renders the HTML with appropriately-localised strings.  Not sure how to solve with server-side rendering at this stage; it seems it would require the pegasus server to talk directly to the rails server.

**Comment by Bjvanminnen** [comment]
I suspect we're going to increasingly hit cases where we dislike that we have different LOC pipelines (I recently realized that courses in teacher-dashboard have different LOC that the same courses in dashboard). Not sure of solutions yet, but something we'll probably want to continue to be cognizant of.

**Comment by breville** [comment]
@Bjvanminnen Yes, that is definitely true.  My understanding is that we prefer the apps & dashboard loc pipelines over the pegasus one, another good reason to build components in React in apps.  @tanyaparker and @Hamms can confirm.

File: pegasus/sites.v3/code.org/public/hourofcode/overview.haml (hunk 1 above)
Comment (Erin007) [reply pegasus/sites.v3/code.org/public/hourofcode/overview.haml]: I changed the width, does she still look stretched to you?

**Comment by Erin007** [comment]
@Bjvanminnen  @breville  Sounds like the consensus is that the way I've done this is less than ideal, but given both time and that the issue is part of a larger, tangled problem regarding dashboard-pegasus split, I can keep it as is?

**Comment by tanyaparker** [comment]
yes. dashboard loc pipeline preferred.

File: pegasus/sites.v3/code.org/public/hourofcode/overview.haml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+%h1 Hour of Code for all ages
+
+.hoc-tiles-container
+
+.full-resource-block
+  %img.side-image.left.col-50-student{src: "/images/hoc_overview.png"}
+  .side-description-area.right.col-50-student
+    .description
+      %h2.resource-title More Activities from Partners
+      The Hour of Code website has over 100 computer science activities from Code.org and partners for all ages and experience levels.
```
Comment (tanyaparker) [inline pegasus/sites.v3/code.org/public/hourofcode/overview.haml]: Sorry this read funny like the partners were for all ages. Can we change to this?

`The Hour of Code website has over 100 computer science activities for all ages and experience levels, developed by Code.org and partners.`

Comment (Erin007) [reply pegasus/sites.v3/code.org/public/hourofcode/overview.haml]: ✅

**Pull request merged**